woocommerce/class-sv-wc-plugin.php

```diff
@@ -106,13 +111,13 @@ abstract class SV_WC_Plugin {
 	 *
 	 * @param string $id plugin id
 	 * @param string $version plugin version number
-	 * @param array $args {
+	 * @param array<string, mixed> $args {
 	 *     optional plugin arguments
 	 *
 	 *     @type int|float $latest_wc_versions the last supported versions of WooCommerce, as a major.minor float relative to the latest available version
 	 *     @type string $text_domain the plugin textdomain, used to set up translations
-	 *     @type bool $supports_hpos whether the plugin supports HPOS (default false)
-	 *     @type array  $dependencies {
+	 *     @type array<string, mixed> $supported_features supported WooCommerce features (HPOS, Blocks, etc.)
+	 *     @type array<string, array<string, mixed>>  $dependencies {
```

Comment:
What do you think about using an array shape do describe the arguments, so that we'd have IDE autocompletion support as well as PHPStan static type checking support?

Whenever I need to update the plugin constructor arguments in individual plugins, I find that I need to go back to the abstract constructor to figure out what it accepts. And even so, it's not clear which "features" I can declare support for.

```php
/**
 * @param array{
 *     latest_wc_versions?: int|float,
 *     text_domain?: string,
 *     supported_features?: array{
 *         hpos?: bool,
 *         blocks?: array{
 *             cart?: bool,
 *             checkout?: bool
 *         }
 *     },
 *     dependencies?: array{
 *         php_extensions?: array<string, mixed>,
 *         php_functions?: array<string, mixed>,
 *         php_settings?: array<string, mixed>
 *     }
 * } $args
 */
```

We could define the various parts as local type aliases`@phpstan-type` in the class doblock and then reuse them, when needed, for example:

```php
/**
 * @phpstan-type SupportedFeatures array{ hpos?: bool, blocks?: array{ cart?: bool, checkout?: bool }}
 */
abstract class SV_WC_Plugin {

	/** @var SupportedFeatures */
	protected $supported_features = [];

	/**
	 * @return SupportedFeatures
	 */
	public function get_supported_features() : array {
		return $this->supported_features;
	}
}
```